Running and profiling on AAC

xenshinu opened the pull request.

gsitaram committed:
Restructure minimap2 code to split seed, chain, and align steps to enable easy integration of GPU kernels

joydddd committed:
add gpu code

xenshinu committed:
include cJson

joydddd committed:
Add gpu kernel and memory pool for each batch

joydddd committed:
Add k memory pool leakage check, TODO: init_stream_gpu() called more than once

joydddd committed:
Fix gpu initialization

joydddd committed:
gpu kernel running, memory pool no leakage. Error output?

joydddd committed:
Clean debug prints

joydddd committed:
Fix Pending batch issue

xenshinu committed:
Merge branch 'gpu_kernel' of https://github.com/Minimap2onGPU/minimap2 into gpu_kernel

joydddd committed:
Merge pull request #6 from Minimap2onGPU/gpu_kernel

Integrate stand alone gpu kernels

joydddd committed:
Merge pull request #5 from Minimap2onGPU/split_kernels

Restructure minimap2 code for batched chaining

xenshinu committed:
add gpu config path as an option

joydddd committed:
Fix long-short kernel; add chaining timer

joydddd committed:
Seperate gpu initialization and free

xenshinu committed:
Merge branch 'develop' into gpu_kernel

xenshinu committed:
Merge branch 'gpu_kernel' of https://github.com/Minimap2onGPU/minimap2 into gpu_kernel

xenshinu committed:
cuda flags

joydddd committed:
Change sc grid calculation

xenshinu committed:
add long seg

xenshinu committed:
merge grid size

xenshinu committed:
change plmem and plscore to use new config

xenshinu committed:
merge with cuda

xenshinu committed:
disable new offset feature

xenshinu committed:
add compile time launch bound

xenshinu committed:
add compile time launch bound using template

joydddd committed:
Add short-mid-long kernel

joydddd committed:
Merge branch 'gpu_kernel' of github.com:Minimap2onGPU/minimap2 into gpu_kernel

joydddd committed:
Add num segs printout

xenshinu committed:
add template to mid kernel launch

xenshinu committed:
merge makefile with nvcc version

joydddd committed:
Add anchor compression

joydddd committed:
Merge branch 'gpu_kernel' of github.com:Minimap2onGPU/minimap2 into gpu_kernel

xenshinu committed:
add global config

xenshinu committed:
merge with independent mid kernel

xenshinu committed:
add cudacheck and uncomment syncthreads

xenshinu committed:
add profile scripts

xenshinu committed:
add profile scripts

xenshinu committed:
add profile scripts

xenshinu committed:
update parseing profile scripts

xenshinu committed:
add timer by event recorder

xenshinu committed:
remove const keyword

xenshinu committed:
merge with aac code including CUDA timer

joydddd committed:
merge with aac code including CUDA timer (#9)

joydddd committed:
merge with aac code including CUDA timer (#9)

xenshinu committed:
add profiling scripts, please check scripts/*.slurm

xenshinu committed:
Merge branch 'gpu_kernel' of github.com:Minimap2onGPU/minimap2 into gpu_kernel

xenshinu committed:
add sample slurm

Luke20000429 commented:
To run on AAC, refer to samples in `scripts/*.slurm`
Please create two folders in workspace `rocprof_output` and `slurm_output`
Submit workload with `sbatch ${name}.slurm`
Check workload queue with `squeue -u $USER`

joydddd committed:
Delete gpu/.depend

joydddd committed:
Fix debug functions

joydddd committed:
Merge commit '1e8781a' into gpu_kernel

xenshinu committed:
add new api translation, enable analysis print

joydddd committed:
Add aggregate long segs (#12)

xenshinu committed:
add omnitrace scripts

xenshinu committed:
finish minibatch, parameter is still hardcoded, debug function need fix, f_long is not copied back

xenshinu committed:
no reset long seg on each micro batch cause fault

joydddd committed:
Add acc_config. FIX seg fault for long_seg_count reset

xenshinu committed:
finish microbatch design, TODO: add batch number to config, and use hostmalloc

xenshinu committed:
use hostmalloc to avoid step1 delay

joydddd committed:
change script path

joydddd committed:
Add kernel throughput calculatation

xenshinu committed:
update scripts

xenshinu committed:
add sorting technique

joydddd committed:
Update throughput calculation, a6000 config

xenshinu committed:
add atomic runtime balancing

xenshinu committed:
Merge branch 'gpu_kernel-break' of github.com:Minimap2onGPU/minimap2 into gpu_kernel-break

xenshinu committed:
debug info control

joydddd committed:
Update debug analysis

joydddd committed:
Edit throughput calculation. JIT Compilat error on cuda, push to try on HIP

joydddd committed:
Fix throughput analysis

xenshinu committed:
fix atomicadd -> atomicsub, TODO: add more cudaCheck

xenshinu committed:
fix atomic add in long seg, only first thread in block add the atomic

joydddd committed:
Temporal Fix microbacthing error (Use CPU kernel)

joydddd committed:
Add put long segs back to original reads, but output seems to be wrong??

xenshinu committed:
add seg count

xenshinu committed:
Merge branch 'gpu_kernel-break' of github.com:Minimap2onGPU/minimap2 into gpu_kernel-break

joydddd committed:
Remove skip backtracking in GPU implementation. Outputs are correct

xenshinu committed:
update plscore

xenshinu committed:
Merge branch 'gpu_kernel-break' of github.com:Minimap2onGPU/minimap2 into gpu_kernel-break

xenshinu committed:
comment in kernel print to maximize tp

xenshinu committed:
config aac that maximize memory usage

joydddd committed:
Add data analysis script

joydddd committed:
Add data analysis script

xenshinu committed:
use less branch compute sc

joydddd committed:
Add range distribution analysis

joydddd committed:
Merge branch 'gpu_kernel-break' of github.com:Minimap2onGPU/minimap2 into gpu_kernel-break

joydddd committed:
cleanup gpu code for open source. TODO: Add README

joydddd committed:
Update print compile time config

joydddd committed:
Merge pull request #13 from Minimap2onGPU:gpu_kernel-break

Add long segment aggregation. Cleaned up code for sysbio submission. TODO: Add README

joydddd committed:
Clean up compile options. Move config to cmd option / gpu_config.json

joydddd committed:
Merge branch 'gpu_kernel-break' into gpu_kernel

joydddd committed:
Add planalyze.cu(h)

xenshinu committed:
latest gpu config

xenshinu committed:
merge with mi210 code

xenshinu committed:
add configs

xenshinu committed:
remove slurms

joydddd committed:
Fix cuda score generation kernel (#14)

* Merge gpu-kernel. Squashed commit of the following:

commit 41062d0195346a9de0ac4dad6ed98dac08177e50
Author: joydddd <joydong@umich.edu>
Date:   Mon Mar 18 13:17:15 2024 -0400

    Clean up compile options. Move config to cmd option / gpu_config.json

commit 32188737df3e857cfa93e1f7f2b919ad1438be01
Author: joydddd <joydong@umich.edu>
Date:   Wed Mar 13 23:31:16 2024 -0400

    Update print compile time config

commit ad70b6a3c9675a7ea26630470407ac54a52f1083
Author: joydddd <joydong@umich.edu>
Date:   Wed Mar 13 23:09:51 2024 -0400

    cleanup gpu code for open source. TODO: Add README

commit 4d2459c8dc876b5ca71bfba803b5b3824795c87c
Merge: 13b2eab 5c742d9
Author: Joy Dong <joydong@umich.edu>
Date:   Wed Mar 13 19:34:56 2024 -0500

    Merge branch 'gpu_kernel-break' of github.com:Minimap2onGPU/minimap2 into gpu_kernel-break

commit 13b2eab52e300d1352af7b3ee097f8881e1881ac
Author: Joy Dong <joydong@umich.edu>
Date:   Wed Mar 13 19:34:05 2024 -0500

    Add range distribution analysis

commit f7ecde1ed2a3cf55f37f01daa9bac57e3dbc8b7f
Author: Joy Dong <joydong@umich.edu>
Date:   Wed Mar 6 12:14:26 2024 -0600

    Add data analysis script

commit 5c742d9cf3eaece56876073f42de2c3ae458b770
Author: Joy Dong <joydong@umich.edu>
Date:   Wed Mar 6 12:14:26 2024 -0600

    Add data analysis script

commit 12c3fc0f0dea2a133e49745d4b9209ca71959daa
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Fri Mar 1 22:50:35 2024 -0600

    config aac that maximize memory usage

commit eed2640686d253b1ca0494171198c6cb54e2caf9
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Fri Mar 1 00:24:31 2024 -0600

    comment in kernel print to maximize tp

commit c982bb556b7fbd93918b70292f8de1e1946df1ab
Merge: 94f333a d8ba447
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Wed Feb 28 11:14:20 2024 -0600

    Merge branch 'gpu_kernel-break' of github.com:Minimap2onGPU/minimap2 into gpu_kernel-break

commit 94f333a8812c257150b7c1a4ba3ecff0021b7102
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Wed Feb 28 11:14:16 2024 -0600

    update plscore

commit d8ba447ae4b88563f78b9a43546dd6f517089010
Author: joydddd <joydong@umich.edu>
Date:   Mon Feb 26 12:40:55 2024 -0500

    Remove skip backtracking in GPU implementation. Outputs are correct

commit 6836cb74d10dc69658b8b944fb7459509b9b4873
Merge: 907748f 3239112
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Fri Feb 23 19:13:51 2024 -0600

    Merge branch 'gpu_kernel-break' of github.com:Minimap2onGPU/minimap2 into gpu_kernel-break

commit 907748f2c12ba0d326d987bcd135cbfbbcdabad6
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Fri Feb 23 19:13:46 2024 -0600

    add seg count

commit 3239112c0e90f137111a6f51824599c6ef8d10d2
Author: joydddd <joydong@umich.edu>
Date:   Thu Feb 22 23:26:47 2024 -0500

    Add put long segs back to original reads, but output seems to be wrong??

commit 474e746a811e7600683caca25eaa7956b791af9a
Author: joydddd <joydong@umich.edu>
Date:   Thu Feb 22 22:37:27 2024 -0500

    Temporal Fix microbacthing error (Use CPU kernel)

commit 66e2e54062be026fe35abbffff602e35a5f3962b
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Thu Feb 22 11:15:52 2024 -0600

    fix atomic add in long seg, only first thread in block add the atomic

commit c40bf9f7ead5dd10192fa7fc6d87b015f579e2bf
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Sat Feb 17 00:38:19 2024 -0600

    fix atomicadd -> atomicsub, TODO: add more cudaCheck

commit 9678399b11792bd8d6f1f447e9b165e650a729ef
Author: Joy Dong <joydong@umich.edu>
Date:   Thu Feb 15 23:45:36 2024 -0600

    Fix throughput analysis

commit 39f758b99c981c1addd07e91be7bce25937dcca1
Author: Joy Dong <joydong@umich.edu>
Date:   Thu Feb 15 22:50:52 2024 -0500

    Edit throughput calculation. JIT Compilat error on cuda, push to try on HIP

commit b21c5f92407e2c632c896082ededb52a8a5e868c
Author: Joy Dong <joydong@umich.edu>
Date:   Tue Feb 13 15:46:20 2024 -0500

    Update debug analysis

commit cff2e270080d47a6efee72646dfc60c36f860ebf
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Fri Feb 9 10:12:52 2024 -0600

    debug info control

commit accea337aba1fec66888c58ae11780fd40849a4d
Merge: d9e7396 e1248fb
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Fri Feb 9 09:44:56 2024 -0600

    Merge branch 'gpu_kernel-break' of github.com:Minimap2onGPU/minimap2 into gpu_kernel-break

commit d9e739603c11baa8a4815248353dd417a8b3ee38
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Fri Feb 9 09:44:52 2024 -0600

    add atomic runtime balancing

commit e1248fbcbde855318f6ea9cb858d72de546cc8a5
Author: Joy Dong <joydong@umich.edu>
Date:   Thu Feb 8 20:01:02 2024 -0500

    Update throughput calculation, a6000 config

commit 4aeacfdacae55bbb1fa4a4825fb49a1d8149f0af
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Thu Feb 8 12:48:15 2024 -0600

    add sorting technique

commit 8ee89c13fb01ff1282e36a4ffd2a5727c3333723
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Tue Feb 6 15:09:15 2024 -0600

    update scripts

commit 216f2b2510931bf75f920e3602b2fcfa4c5ba953
Author: Joy Dong <joydong@umich.edu>
Date:   Fri Jan 26 12:01:24 2024 -0600

    Add kernel throughput calculatation

commit 15d0003c30aa85c49a544bd87185175bf866cbb6
Author: Joy Dong <joydong@umich.edu>
Date:   Wed Jan 10 21:11:34 2024 -0600

    change script path

commit aa626439f9dce504aac634a1f0b8044349e23d2c
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Wed Jan 10 14:37:39 2024 -0600

    use hostmalloc to avoid step1 delay

commit 85f1cbb053a90b466a4d22c101730f95e94f48fe
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Fri Nov 3 19:26:00 2023 -0500

    finish microbatch design, TODO: add batch number to config, and use hostmalloc

commit 1a585abd9263fe927a13657257886121fd835646
Author: Joy Dong <joydong@umich.edu>
Date:   Mon Oct 30 16:23:50 2023 -0500

    Add acc_config. FIX seg fault for long_seg_count reset

commit cb1a30eb4362d8b7fdfa3a5b5865f7135f66bbae
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Fri Oct 27 13:56:12 2023 -0500

    no reset long seg on each micro batch cause fault

commit 5e1abe8a05f7e2e2f1a39515b6d25987cc2c2135
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Fri Oct 27 11:34:49 2023 -0500

    finish minibatch, parameter is still hardcoded, debug function need fix, f_long is not copied back

commit 559871804e183edd8a920bb436df765b8ba563e4
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Thu Oct 26 09:46:48 2023 -0500

    add omnitrace scripts

commit 31dff82991479f19f8b4ad887bc9663656e7584f
Author: Joy Juechu Dong <joydong@umich.edu>
Date:   Thu Oct 19 23:01:03 2023 -0400

    Add aggregate long segs (#12)

commit cd225fa5a0f065f5cff7fae5a78b7d8d1d799c70
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Thu Oct 12 09:49:48 2023 -0500

    add new api translation, enable analysis print

commit 247a17f3f04b0dcfc53b209fcaf342a0f94a2c01
Merge: 1d5c636 1e8781a
Author: joydddd <joydong@umich.edu>
Date:   Tue Oct 10 22:00:31 2023 -0400

    Merge commit '1e8781a' into gpu_kernel

commit 1e8781aecdfa7f044d900d6526a1f47b6252b060
Author: joydddd <joydong@umich.edu>
Date:   Tue Oct 10 21:56:18 2023 -0400

    Fix debug functions

commit 1d5c6362f75a506678c624611d331b8fff18b902
Merge: d334c9e ccddf5c
Author: joydddd <joydong@umich.edu>
Date:   Wed Sep 27 11:22:11 2023 -0400

    Delete gpu/.depend

commit d334c9e5fb0bf0ca90643b1e37175e91a2fec73f
Author: Joy Juechu Dong <joydong@umich.edu>
Date:   Tue Sep 26 15:30:12 2023 -0400

    merge with aac code including CUDA timer (#9)

commit ccddf5c3f4b6641ecce3a18291935569aeb3792d
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Tue Sep 26 16:55:27 2023 -0500

    add sample slurm

commit 7b65403dadb0fc75f2e195b78f2372deca271d20
Merge: cabe321 6897137
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Tue Sep 26 16:41:38 2023 -0500

    Merge branch 'gpu_kernel' of github.com:Minimap2onGPU/minimap2 into gpu_kernel

commit cabe3211773f8445e36ad26bf09c6a0c26b7d8fe
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Tue Sep 26 16:39:58 2023 -0500

    add profiling scripts, please check scripts/*.slurm

commit 6897137979d5d16c60548829d43569a383b47646
Author: Joy Juechu Dong <joydong@umich.edu>
Date:   Tue Sep 26 15:30:12 2023 -0400

    merge with aac code including CUDA timer (#9)

commit e144d2d68525d311cfff613e3cba4dfae25a9038
Merge: 4044b9f 1d5d7ab
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Tue Sep 5 19:50:37 2023 -0400

    merge with aac code including CUDA timer

commit 4044b9f63cfb1626c13ef0168e2dfb404a16b859
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Tue Sep 5 19:47:31 2023 -0400

    remove const keyword

commit 1d5d7ab117af889fca030170abd4863e9c3724a4
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Tue Sep 5 17:52:39 2023 -0500

    add timer by event recorder

commit 10e71977e53f6f121ed1a0133672b8170cd7cd78
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Wed Jul 12 10:49:27 2023 -0400

    update parseing profile scripts

commit ae3ecd3f32bed4a4b1abacb6722d895b417ddfe1
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Wed Jul 12 10:46:53 2023 -0400

    add profile scripts

commit cff2019c77bc4d75e00f6975418c0cbdca857015
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Wed Jul 12 10:46:41 2023 -0400

    add profile scripts

commit 26a404433aa4b204b005b55e90fa9f2e78aeb141
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Wed Jul 12 10:45:37 2023 -0400

    add profile scripts

commit ced508a1a223a72eec69054e4785b604fec89281
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Tue Jul 11 22:47:19 2023 -0400

    add cudacheck and uncomment syncthreads

commit a01dd39c2e6c52d729eda4d7918284821ba357ab
Merge: ea9c438 10a5a8b
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Tue Jul 11 16:03:45 2023 -0400

    merge with independent mid kernel

commit ea9c43839743c63413f4f554b6430db62eaaa98a
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Tue Jul 11 15:19:51 2023 -0400

    add global config

commit 10a5a8b035f9073d2358e1d5af9f66ef2db04fec
Merge: 7018ece 18bc4ac
Author: joydddd <joydong@umich.edu>
Date:   Wed Jul 5 11:06:17 2023 -0400

    Merge branch 'gpu_kernel' of github.com:Minimap2onGPU/minimap2 into gpu_kernel

commit 7018ece4ee03424e0a7d12c99274ba40f09cd723
Author: joydddd <joydong@umich.edu>
Date:   Wed Jul 5 10:58:35 2023 -0400

    Add anchor compression

commit 18bc4accb887cace973e72f6749ae2a8d360db31
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Wed Jun 28 19:21:06 2023 -0400

    merge makefile with nvcc version

commit 5a12d5b37e11f1b4cb9eb4c0d5269669f5b0b692
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Wed Jun 28 19:19:25 2023 -0400

    add template to mid kernel launch

commit 43180787b09e1d5b75026914975ce0739715f1b8
Author: joydddd <joydong@umich.edu>
Date:   Wed Jun 28 15:28:53 2023 -0400

    Add num segs printout

commit 5906e297bfd86629ea2be21430575787fd8771d7
Merge: 9149e17 e6f016c
Author: joydddd <joydong@umich.edu>
Date:   Wed Jun 28 13:17:41 2023 -0400

    Merge branch 'gpu_kernel' of github.com:Minimap2onGPU/minimap2 into gpu_kernel

commit 9149e17ee1784142bce8b742d4c4dfc0681b3cb1
Author: joydddd <joydong@umich.edu>
Date:   Wed Jun 28 13:11:40 2023 -0400

    Add short-mid-long kernel

commit e6f016c619f95b6c91ffeb676004996448481b95
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Sun Jun 25 12:21:50 2023 -0400

    add compile time launch bound using template

commit c601518ca0649a091b2eac117361680545f18b5b
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Sun Jun 25 12:15:49 2023 -0400

    add compile time launch bound

commit be01974322e412b6262d5d955790613e910d6033
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Sat Jun 10 16:16:23 2023 -0400

    disable new offset feature

commit 2d464996dce2c3fb63643ce2a14648a148ca99b1
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Sat Jun 10 16:05:53 2023 -0400

    merge with cuda

* Revert "Merge gpu-kernel. Squashed commit of the following:"

This reverts commit 481f3a5d86271e64aff0a237c6f2ce9cf7360ac1.

* Merge gpu_kernel
Squashed commit of the following:

commit 938a45f67f0886b6ca00609d5b898eb9dac5835c
Author: joydddd <joydong@umich.edu>
Date:   Mon Mar 18 13:26:25 2024 -0400

    Add planalyze.cu(h)

commit fee935ffdd5d5a1f990b51cb671b3b3884b2db41
Merge: ec8a340 41062d0
Author: joydddd <joydong@umich.edu>
Date:   Mon Mar 18 13:18:42 2024 -0400

    Merge branch 'gpu_kernel-break' into gpu_kernel

commit 41062d0195346a9de0ac4dad6ed98dac08177e50
Author: joydddd <joydong@umich.edu>
Date:   Mon Mar 18 13:17:15 2024 -0400

    Clean up compile options. Move config to cmd option / gpu_config.json

commit ec8a34086db6e47bc2e88295b9cd9935aad6ad3c
Merge: 31dff82 3218873
Author: Joy Juechu Dong <joydong@umich.edu>
Date:   Wed Mar 13 23:37:55 2024 -0400

    Merge pull request #13 from Minimap2onGPU:gpu_kernel-break

    Add long segment aggregation. Cleaned up code for sysbio submission. TODO: Add README

commit 32188737df3e857cfa93e1f7f2b919ad1438be01
Author: joydddd <joydong@umich.edu>
Date:   Wed Mar 13 23:31:16 2024 -0400

    Update print compile time config

commit ad70b6a3c9675a7ea26630470407ac54a52f1083
Author: joydddd <joydong@umich.edu>
Date:   Wed Mar 13 23:09:51 2024 -0400

    cleanup gpu code for open source. TODO: Add README

commit 4d2459c8dc876b5ca71bfba803b5b3824795c87c
Merge: 13b2eab 5c742d9
Author: Joy Dong <joydong@umich.edu>
Date:   Wed Mar 13 19:34:56 2024 -0500

    Merge branch 'gpu_kernel-break' of github.com:Minimap2onGPU/minimap2 into gpu_kernel-break

commit 13b2eab52e300d1352af7b3ee097f8881e1881ac
Author: Joy Dong <joydong@umich.edu>
Date:   Wed Mar 13 19:34:05 2024 -0500

    Add range distribution analysis

commit f7ecde1ed2a3cf55f37f01daa9bac57e3dbc8b7f
Author: Joy Dong <joydong@umich.edu>
Date:   Wed Mar 6 12:14:26 2024 -0600

    Add data analysis script

commit 5c742d9cf3eaece56876073f42de2c3ae458b770
Author: Joy Dong <joydong@umich.edu>
Date:   Wed Mar 6 12:14:26 2024 -0600

    Add data analysis script

commit 12c3fc0f0dea2a133e49745d4b9209ca71959daa
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Fri Mar 1 22:50:35 2024 -0600

    config aac that maximize memory usage

commit eed2640686d253b1ca0494171198c6cb54e2caf9
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Fri Mar 1 00:24:31 2024 -0600

    comment in kernel print to maximize tp

commit c982bb556b7fbd93918b70292f8de1e1946df1ab
Merge: 94f333a d8ba447
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Wed Feb 28 11:14:20 2024 -0600

    Merge branch 'gpu_kernel-break' of github.com:Minimap2onGPU/minimap2 into gpu_kernel-break

commit 94f333a8812c257150b7c1a4ba3ecff0021b7102
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Wed Feb 28 11:14:16 2024 -0600

    update plscore

commit d8ba447ae4b88563f78b9a43546dd6f517089010
Author: joydddd <joydong@umich.edu>
Date:   Mon Feb 26 12:40:55 2024 -0500

    Remove skip backtracking in GPU implementation. Outputs are correct

commit 6836cb74d10dc69658b8b944fb7459509b9b4873
Merge: 907748f 3239112
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Fri Feb 23 19:13:51 2024 -0600

    Merge branch 'gpu_kernel-break' of github.com:Minimap2onGPU/minimap2 into gpu_kernel-break

commit 907748f2c12ba0d326d987bcd135cbfbbcdabad6
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Fri Feb 23 19:13:46 2024 -0600

    add seg count

commit 3239112c0e90f137111a6f51824599c6ef8d10d2
Author: joydddd <joydong@umich.edu>
Date:   Thu Feb 22 23:26:47 2024 -0500

    Add put long segs back to original reads, but output seems to be wrong??

commit 474e746a811e7600683caca25eaa7956b791af9a
Author: joydddd <joydong@umich.edu>
Date:   Thu Feb 22 22:37:27 2024 -0500

    Temporal Fix microbacthing error (Use CPU kernel)

commit 66e2e54062be026fe35abbffff602e35a5f3962b
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Thu Feb 22 11:15:52 2024 -0600

    fix atomic add in long seg, only first thread in block add the atomic

commit c40bf9f7ead5dd10192fa7fc6d87b015f579e2bf
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Sat Feb 17 00:38:19 2024 -0600

    fix atomicadd -> atomicsub, TODO: add more cudaCheck

commit 9678399b11792bd8d6f1f447e9b165e650a729ef
Author: Joy Dong <joydong@umich.edu>
Date:   Thu Feb 15 23:45:36 2024 -0600

    Fix throughput analysis

commit 39f758b99c981c1addd07e91be7bce25937dcca1
Author: Joy Dong <joydong@umich.edu>
Date:   Thu Feb 15 22:50:52 2024 -0500

    Edit throughput calculation. JIT Compilat error on cuda, push to try on HIP

commit b21c5f92407e2c632c896082ededb52a8a5e868c
Author: Joy Dong <joydong@umich.edu>
Date:   Tue Feb 13 15:46:20 2024 -0500

    Update debug analysis

commit cff2e270080d47a6efee72646dfc60c36f860ebf
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Fri Feb 9 10:12:52 2024 -0600

    debug info control

commit accea337aba1fec66888c58ae11780fd40849a4d
Merge: d9e7396 e1248fb
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Fri Feb 9 09:44:56 2024 -0600

    Merge branch 'gpu_kernel-break' of github.com:Minimap2onGPU/minimap2 into gpu_kernel-break

commit d9e739603c11baa8a4815248353dd417a8b3ee38
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Fri Feb 9 09:44:52 2024 -0600

    add atomic runtime balancing

commit e1248fbcbde855318f6ea9cb858d72de546cc8a5
Author: Joy Dong <joydong@umich.edu>
Date:   Thu Feb 8 20:01:02 2024 -0500

    Update throughput calculation, a6000 config

commit 4aeacfdacae55bbb1fa4a4825fb49a1d8149f0af
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Thu Feb 8 12:48:15 2024 -0600

    add sorting technique

commit 8ee89c13fb01ff1282e36a4ffd2a5727c3333723
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Tue Feb 6 15:09:15 2024 -0600

    update scripts

commit 216f2b2510931bf75f920e3602b2fcfa4c5ba953
Author: Joy Dong <joydong@umich.edu>
Date:   Fri Jan 26 12:01:24 2024 -0600

    Add kernel throughput calculatation

commit 15d0003c30aa85c49a544bd87185175bf866cbb6
Author: Joy Dong <joydong@umich.edu>
Date:   Wed Jan 10 21:11:34 2024 -0600

    change script path

commit aa626439f9dce504aac634a1f0b8044349e23d2c
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Wed Jan 10 14:37:39 2024 -0600

    use hostmalloc to avoid step1 delay

commit 85f1cbb053a90b466a4d22c101730f95e94f48fe
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Fri Nov 3 19:26:00 2023 -0500

    finish microbatch design, TODO: add batch number to config, and use hostmalloc

commit 1a585abd9263fe927a13657257886121fd835646
Author: Joy Dong <joydong@umich.edu>
Date:   Mon Oct 30 16:23:50 2023 -0500

    Add acc_config. FIX seg fault for long_seg_count reset

commit cb1a30eb4362d8b7fdfa3a5b5865f7135f66bbae
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Fri Oct 27 13:56:12 2023 -0500

    no reset long seg on each micro batch cause fault

commit 5e1abe8a05f7e2e2f1a39515b6d25987cc2c2135
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Fri Oct 27 11:34:49 2023 -0500

    finish minibatch, parameter is still hardcoded, debug function need fix, f_long is not copied back

commit 559871804e183edd8a920bb436df765b8ba563e4
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Thu Oct 26 09:46:48 2023 -0500

    add omnitrace scripts

commit 31dff82991479f19f8b4ad887bc9663656e7584f
Author: Joy Juechu Dong <joydong@umich.edu>
Date:   Thu Oct 19 23:01:03 2023 -0400

    Add aggregate long segs (#12)

commit cd225fa5a0f065f5cff7fae5a78b7d8d1d799c70
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Thu Oct 12 09:49:48 2023 -0500

    add new api translation, enable analysis print

commit 247a17f3f04b0dcfc53b209fcaf342a0f94a2c01
Merge: 1d5c636 1e8781a
Author: joydddd <joydong@umich.edu>
Date:   Tue Oct 10 22:00:31 2023 -0400

    Merge commit '1e8781a' into gpu_kernel

commit 1e8781aecdfa7f044d900d6526a1f47b6252b060
Author: joydddd <joydong@umich.edu>
Date:   Tue Oct 10 21:56:18 2023 -0400

    Fix debug functions

commit 1d5c6362f75a506678c624611d331b8fff18b902
Merge: d334c9e ccddf5c
Author: joydddd <joydong@umich.edu>
Date:   Wed Sep 27 11:22:11 2023 -0400

    Delete gpu/.depend

commit d334c9e5fb0bf0ca90643b1e37175e91a2fec73f
Author: Joy Juechu Dong <joydong@umich.edu>
Date:   Tue Sep 26 15:30:12 2023 -0400

    merge with aac code including CUDA timer (#9)

commit ccddf5c3f4b6641ecce3a18291935569aeb3792d
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Tue Sep 26 16:55:27 2023 -0500

    add sample slurm

commit 7b65403dadb0fc75f2e195b78f2372deca271d20
Merge: cabe321 6897137
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Tue Sep 26 16:41:38 2023 -0500

    Merge branch 'gpu_kernel' of github.com:Minimap2onGPU/minimap2 into gpu_kernel

commit cabe3211773f8445e36ad26bf09c6a0c26b7d8fe
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Tue Sep 26 16:39:58 2023 -0500

    add profiling scripts, please check scripts/*.slurm

commit 6897137979d5d16c60548829d43569a383b47646
Author: Joy Juechu Dong <joydong@umich.edu>
Date:   Tue Sep 26 15:30:12 2023 -0400

    merge with aac code including CUDA timer (#9)

commit e144d2d68525d311cfff613e3cba4dfae25a9038
Merge: 4044b9f 1d5d7ab
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Tue Sep 5 19:50:37 2023 -0400

    merge with aac code including CUDA timer

commit 4044b9f63cfb1626c13ef0168e2dfb404a16b859
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Tue Sep 5 19:47:31 2023 -0400

    remove const keyword

commit 1d5d7ab117af889fca030170abd4863e9c3724a4
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Tue Sep 5 17:52:39 2023 -0500

    add timer by event recorder

commit 10e71977e53f6f121ed1a0133672b8170cd7cd78
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Wed Jul 12 10:49:27 2023 -0400

    update parseing profile scripts

commit ae3ecd3f32bed4a4b1abacb6722d895b417ddfe1
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Wed Jul 12 10:46:53 2023 -0400

    add profile scripts

commit cff2019c77bc4d75e00f6975418c0cbdca857015
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Wed Jul 12 10:46:41 2023 -0400

    add profile scripts

commit 26a404433aa4b204b005b55e90fa9f2e78aeb141
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Wed Jul 12 10:45:37 2023 -0400

    add profile scripts

commit ced508a1a223a72eec69054e4785b604fec89281
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Tue Jul 11 22:47:19 2023 -0400

    add cudacheck and uncomment syncthreads

commit a01dd39c2e6c52d729eda4d7918284821ba357ab
Merge: ea9c438 10a5a8b
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Tue Jul 11 16:03:45 2023 -0400

    merge with independent mid kernel

commit ea9c43839743c63413f4f554b6430db62eaaa98a
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Tue Jul 11 15:19:51 2023 -0400

    add global config

commit 10a5a8b035f9073d2358e1d5af9f66ef2db04fec
Merge: 7018ece 18bc4ac
Author: joydddd <joydong@umich.edu>
Date:   Wed Jul 5 11:06:17 2023 -0400

    Merge branch 'gpu_kernel' of github.com:Minimap2onGPU/minimap2 into gpu_kernel

commit 7018ece4ee03424e0a7d12c99274ba40f09cd723
Author: joydddd <joydong@umich.edu>
Date:   Wed Jul 5 10:58:35 2023 -0400

    Add anchor compression

commit 18bc4accb887cace973e72f6749ae2a8d360db31
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Wed Jun 28 19:21:06 2023 -0400

    merge makefile with nvcc version

commit 5a12d5b37e11f1b4cb9eb4c0d5269669f5b0b692
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Wed Jun 28 19:19:25 2023 -0400

    add template to mid kernel launch

commit 43180787b09e1d5b75026914975ce0739715f1b8
Author: joydddd <joydong@umich.edu>
Date:   Wed Jun 28 15:28:53 2023 -0400

    Add num segs printout

commit 5906e297bfd86629ea2be21430575787fd8771d7
Merge: 9149e17 e6f016c
Author: joydddd <joydong@umich.edu>
Date:   Wed Jun 28 13:17:41 2023 -0400

    Merge branch 'gpu_kernel' of github.com:Minimap2onGPU/minimap2 into gpu_kernel

commit 9149e17ee1784142bce8b742d4c4dfc0681b3cb1
Author: joydddd <joydong@umich.edu>
Date:   Wed Jun 28 13:11:40 2023 -0400

    Add short-mid-long kernel

commit e6f016c619f95b6c91ffeb676004996448481b95
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Sun Jun 25 12:21:50 2023 -0400

    add compile time launch bound using template

commit c601518ca0649a091b2eac117361680545f18b5b
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Sun Jun 25 12:15:49 2023 -0400

    add compile time launch bound

commit be01974322e412b6262d5d955790613e910d6033
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Sat Jun 10 16:16:23 2023 -0400

    disable new offset feature

commit 2d464996dce2c3fb63643ce2a14648a148ca99b1
Author: Xueshen Liu <liuxs@umich.edu>
Date:   Sat Jun 10 16:05:53 2023 -0400

    merge with cuda

* FIXME: Warp Illegal Address at __shfl_sync()

* FIXME: seg start_idx end_idx out of range

* Fix cuda score generation kernel

* Fix merge error

joydddd committed:
Update README.md

joydddd committed:
Organize GPU configs

joydddd committed:
Remove additional scripts

harisankarsadasivan committed:
Update README.md

readme updates

harisankarsadasivan committed:
Update README.md

harisankarsadasivan committed:
Update README.md

harisankarsadasivan committed:
Update README.md

joydddd committed:
disable -G -g for DEBUG=analyze to enable CUDA optimization

joydddd committed:
Update a6000 config

joydddd committed:
FIX: NV performance, Update README.md

joydddd committed:
turn off assert for DEUBG<verbose

joydddd committed:
Update README.md, Add citation.
